### PR TITLE
Update pgvector.md

### PR DESCRIPTION
**Pull request by kuraleta**
## Issue

Closes #

## General checklist

- [ ] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 1]
```diff
@@ -17,12 +17,44 @@ https://github.com/pgvector/pgvector
...
+#### Explanation of the Command:<br>
+```docker run```: Runs a new container.<br>
+```--rm```: Automatically removes the container after it stops, ensuring no residual data.<br>
+```--name langchain4j-postgres-test-container```: Names the container langchain4j-postgres-test-container for easy identification.<br>
+```-p 5432:5432```: Maps port 5432 on your local machine to port 5432 in the container.<br>
+```-e POSTGRES_USER=my_user```: Sets the PostgreSQL username to my_user.<br>
+```-e POSTGRES_PASSWORD=my_password```: Sets the PostgreSQL password to my_password.<br>
+```gvector/pgvector```: Specifies the Docker image to use, pre-configured with the PGVector extension.<br>
+
+```
```
Comment (kpavlov) [inline docs/docs/integrations/embedding-stores/pgvector.md:80]:
```suggestion
```java

```

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 2]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+
+```implementation 'dev.langchain4j:langchain4j-pgvector:0.36.2'```
 
 ## APIs
 
 - `PgVectorEmbeddingStore`
 
+## Parameter Summary
+
+| Plain Java Property      | Spring Application Property | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Default Value       | Required/Optional                                                                                                                                                                                                                                                                                |
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]: Let's remove spring column for the time being, it only pollutes the table right now

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 3]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+| `datasource`             | `coming soon`               | The `DataSource` object used for database connections.                                                                                                                                                                                                                                                                                                                                                                                                         | None                | Required if `host`, `port`, `user`, `password`, and `database` are not provided individually.                                                                                                                                                                                                    |
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]: ordering of properties could be improved: first required, then optional. First most important, then less important. `datasource` should be close to `host`/`port`/etc. `table` and `createTale` should be closee to each other, etc

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 4]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+| `metadataStorageConfig`  | `coming soon`               | Configuration object for handling metadata associated with embeddings. Supports three storage modes: <ul><li>**COLUMN_PER_KEY**: For static metadata when you know the metadata keys in advance.</li><li>**COMBINED_JSON**: For dynamic metadata when you don’t know the metadata keys in advance. Stores data as JSON. (Default)</li><li>**COMBINED_JSONB**: Similar to JSON, but stored in binary format for optimized querying on large datasets.</li></ul> | Default configuration using `COMBINED_JSON` mode. | Optional. If not set, a default configuration is used with `COMBINED_JSON`.|
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]:
```suggestion
| `metadataStorageConfig`  | `coming soon`               | Configuration object for handling metadata associated with embeddings. Supports three storage modes: <ul><li>**COLUMN_PER_KEY**: For static metadata when you know the metadata keys in advance.</li><li>**COMBINED_JSON**: For dynamic metadata when you don’t know the metadata keys in advance. Stores data as JSON. (Default)</li><li>**COMBINED_JSONB**: Similar to JSON, but stored in binary format for optimized querying on large datasets.</li></ul> | `COMBINED_JSON` | Optional. If not set, a default configuration is used with `COMBINED_JSON`.|

```

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 5]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+| `indexListSize`          | `coming soon`               | The number of lists for the IVFFlat index. Required only if `useIndex` is `true`.                                                                                                                                                                                                                                                                                                                                                                              | None                | When Required: <br> If `useIndex` is `true`, `indexListSize` must be provided and must be greater than zero. Otherwise, the program will throw an exception during table initialization. <br> When Optional: <br> If `useIndex` is `false`, this property is ignored and doesn’t need to be set. |
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]: info on required for `indexListSize` is a bit duplicated

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 4 above)
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]: info on required for `metadataStorageConfig` is duplicated

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 6]
```diff
@@ -4,7 +4,7 @@ sidebar_position: 18
 
 # PGVector
 
-https://github.com/pgvector/pgvector
+LangChain4j integrates seamlessly with [PGVector](https://github.com/pgvector/pgvector), allowing developers to store and query vector embeddings directly in PostgreSQL. This integration is ideal for applications like semantic search, recommendation systems, and more.
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]:
```suggestion
LangChain4j integrates seamlessly with [PGVector](https://github.com/pgvector/pgvector), allowing developers to store and query vector embeddings directly in PostgreSQL. This integration is ideal for applications like semantic search, RAG, and more.

```

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 7]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+```
+
+#### Explanation of the Command:<br>
+```docker run```: Runs a new container.<br>
+```--rm```: Automatically removes the container after it stops, ensuring no residual data.<br>
+```--name langchain4j-postgres-test-container```: Names the container langchain4j-postgres-test-container for easy identification.<br>
+```-p 5432:5432```: Maps port 5432 on your local machine to port 5432 in the container.<br>
+```-e POSTGRES_USER=my_user```: Sets the PostgreSQL username to my_user.<br>
+```-e POSTGRES_PASSWORD=my_password```: Sets the PostgreSQL password to my_password.<br>
+```gvector/pgvector```: Specifies the Docker image to use, pre-configured with the PGVector extension.<br>
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]:
```suggestion
```pgvector/pgvector```: Specifies the Docker image to use, pre-configured with the PGVector extension.<br>

```

File: docs/docs/integrations/embedding-stores/pgvector.md [hunk 8]
```diff
@@ -17,12 +17,63 @@ https://github.com/pgvector/pgvector
...
+
+```java
+
+// Initialize the PGVector embedding store
+        EmbeddingStore<TextSegment> embeddingStore = PgVectorEmbeddingStore.builder()
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/pgvector.md]: formatting is off

**Review by ghost — COMMENTED** [review]
@kuraleta thanks a lot, great start!

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 2 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: done

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 3 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: done

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 7 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: changed

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 6 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: done

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 8 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: ?

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 5 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: done

File: docs/docs/integrations/embedding-stores/pgvector.md (hunk 4 above)
Comment (kuraleta) [reply docs/docs/integrations/embedding-stores/pgvector.md]: done

**Review by ghost — APPROVED** [review]
@kuraleta thanks a lot, great job!

**Pull request merged**